### PR TITLE
service worker 

### PR DESCRIPTION
**Pull request by whuang001**
- time spent: 8 hours
- why?:
  - set up service work for interface cache and API cache
  - use local storage to store POST responses.
- effects:
  - now you will be able to refresh our website even if offline
  - when offline, you can still make a request to server. The request will sent when you back online.
  - faster UI loading.
- TODOs:
  - modify POST API to use local storage
- Merge conflict:
  - merge this PR  before #99 
- DOC:
  - https://github.com/CSC59939/Rails/wiki/How-to-handle-POST-responses%3F

### CODE REVIEW

File: website/src/index.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ messaging.usePublicVapidKey('BLTf1ScMkukM6eMWQ3c713ZwaJ2Relaal_FWIUjiNf-ztSlDVQW
 
 ReactDOM.render(<FirebaseApp />, document.getElementById('root'));
 
+// Google Service Worker. Support Chrome 64+ and firefox 13+. Tested on Chrome 70.0.3538.110 (64 bit).
+
 // If you want your app to work offline and load faster, you can change
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: http://bit.ly/CRA-PWA
-serviceWorker.unregister();
+serviceWorker.register();
```
Comment (khelif96) [inline website/src/index.js]: new line

File: website/src/views/DashboardRouter/DashboardRouter.jsx [hunk 2]
```diff
@@ -18,33 +18,42 @@ class DashboardRouter extends PureComponent {
...
-              console.log(err);
-            });
-        });
+          });
+      }
+    } else {
+      this.setState({
+        teacher: localStorage.getItem('DashboardRouter-teacher') === 'true',
+        userData: JSON.parse(localStorage.getItem('DashboardRouter-userData')),
+      });
```
Comment (khelif96) [inline website/src/views/DashboardRouter/DashboardRouter.jsx]: NIce

**Comment by khelif96** [comment]
Why is there a merge conflict?

**Comment by whuang001** [comment]
> Why is there a merge conflict?

Ayushya and me both edit the getprofile API. So merge this RP before #99

**Pull request merged**